### PR TITLE
feat(batch): add secrets props to job definition

### LINKED ISSUE #10976: [Batch] Add "Secrets" properties in interface JobDefinitionContainer
### Use Case

Need to pass Secrets to a Batch Job with CDK Batch Construct.
Not available yet : 
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.JobDefinitionContainer.html

### Proposed Solution

Like with CloudFormation, it'll be interesting to add "Secrets" propertie to the CDK Batch Construct : 
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.CfnJobDefinition.ContainerPropertiesProperty.html#secrets

### Other

https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-batch.CfnJobDefinition.SecretProperty.html

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by yoshizawa56**
closes #10976 

Add a `secrets` property to `batch.JobDefinitionContainer`. This interface is almost the same as `ecs.ContainerDefinitionOptions`.

### CODE REVIEW

File: packages/@aws-cdk/aws-batch/lib/job-definition.ts [hunk 1]
```diff
@@ -112,6 +112,13 @@ export interface JobDefinitionContainer {
    */
   readonly environment?: { [key: string]: string };
 
+  /**
+   * The environment variables from secrets manager or ssm parameter store
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-batch/lib/job-definition.ts:116]: `ecs.Secret` will not support SSM parameter store though.

Don't you want `core.SecretValue` ?

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-batch/lib/job-definition.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-batch/lib/job-definition.ts:116]: Well I guess not if you need the ARN. But in any case, the SSM comment here is inaccurate.

Comment (yoshizawa56) [reply packages/@aws-cdk/aws-batch/lib/job-definition.ts:116]: To my understanding, `ecs.Secret` supports SSM parameter store through `fromSsmParameter` method.
Is it correct?

**Comment by pradoz** [comment]
@rix0rrr Anything else to be done here besides updating the comment?

File: packages/@aws-cdk/aws-batch/lib/job-definition.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-batch/lib/job-definition.ts:116]: Sorry, I was thinking of `ssm.Secret`.

**Review by rix0rrr — DISMISSED** [review]

**Comment by yoshizawa56** [comment]
@rix0rrr What should I do next? This is my first OSS Contribution, please tell me.

**Pull request closed without merging**